sklearn/_engine/tests/test_engines.py

```diff
@@ -136,3 +140,32 @@ def test_get_engine_class():
     next(engine_classes)
     with pytest.raises(ImportError, match=re.escape("sklearn.provider1")):
         next(engine_classes)
+
+
+@pytest.mark.parametrize(
+    "attribute_types,converted", [("sklearn_types", True), ("engine_types", False)]
+)
+def test_attribute_conversion(attribute_types, converted):
```

Comment:
I think some older versions of numpy still supported by scikit-learn my not expose the `np.array_api` module and we should skip this test in that case.